be/src/exec/query_cache/cache_operator.cpp

```diff
@@ -268,6 +278,40 @@ void CacheOperator::_handle_stale_cache_value(int64_t tablet_id, CacheValue& cac
     buffer->chunks.back()->owner_info().set_last_chunk(false);
 }
 
+void CacheOperator::_handle_stale_cache_value_for_pk(int64_t tablet_id, starrocks::query_cache::CacheValue& cache_value,
+                                                     starrocks::query_cache::PerLaneBufferPtr& buffer,
+                                                     int64_t version) {
+    DCHECK(_cache_param.keys_type == TKeysType::PRIMARY_KEYS);
+    // At the present, PRIMARY_KEYS can not support merge-on-read, so we can not merge stale cache values and delta
+    // rowsets. Capturing delta rowsets is meaningless and unsupported, thus we capture all rowsets of the PK tablet.
+    auto status = StorageEngine::instance()->tablet_manager()->capture_tablet_and_rowsets(tablet_id, 0, version);
+    if (!status.ok()) {
+        buffer->state = PLBS_MISS;
+        buffer->cached_version = 0;
+        return;
+    }
+    auto& [tablet, rowsets] = status.value();
+    const auto snapshot_version = cache_value.version;
+    bool has_compacted_delta_rowsets = false;
+    bool exists_non_empty_delta_rowsets = false;
+    for (auto& rs : rowsets) {
+        has_compacted_delta_rowsets |= rs->start_version() == snapshot_version + 1;
+        exists_non_empty_delta_rowsets |= rs->start_version() > snapshot_version && rs->has_data_files();
+    }
+    if (exists_non_empty_delta_rowsets && has_compacted_delta_rowsets) {
```

Comment:
Why  `has_compacted_delta_rowsets `  affects? why not `exists_non_empty_delta_rowsets || exists_non_empty_delta_rowsets `?

Reply:
the condition is wrong, fixed